### PR TITLE
Allow arrow keys and enter key button to navigate main menu

### PR DESCRIPTION
**Pull request by jefetienne**
Using the arrow-up key will select the option above, and arrow-down below. Pressing the enter key will activate the currently selected option.

### CODE REVIEW

File: signus/src/mainmenu.cpp [hunk 1]
```diff
@@ -106,8 +106,11 @@ static int ProcessMenu(const char *mask1, const char *mask2) {
 
 	DrawMN(pt1, pt2, bg, sel);
 
-	while (TRUE) {
+	int run = TRUE;
```
Comment (nextghost) [inline signus/src/mainmenu.cpp:109]: Move this up to the other variable declarations.

File: signus/src/mainmenu.cpp [hunk 2]
```diff
@@ -122,25 +125,44 @@ static int ProcessMenu(const char *mask1, const char *mask2) {
...
-			break;
-		}
-
-		if ((e.What == evKeyDown) && (e.Key.KeyCode == kbEsc)) {
+		} else if (e.What == evMouseDown) {
+			action = 2; // Flash
+		} else if (e.What == evKeyDown) {
+			switch (e.Key.KeyCode) {
+				// Flash
+				case kbEnter:
```
Comment (nextghost) [inline signus/src/mainmenu.cpp:135]: Please keep `case` statements on the same indent level as the parent `switch`. Indent only the code in each `case`.

File: signus/src/mainmenu.cpp [hunk 3]
```diff
@@ -122,25 +125,44 @@ static int ProcessMenu(const char *mask1, const char *mask2) {
...
-		if (e.What == evQuit) {
-			PutEvent(&e);
-			sel = 3;
-			break;
+		if (action == 1) {
+			PlaySample(MenuSnd, 8, 32, 128);
+			DrawMN(pt1, pt2, bg, sel);
+		} else if (action == 2) {
+			Flash(80, 250 + 77 * sel);
+			run = FALSE;
```
Comment (nextghost) [inline signus/src/mainmenu.cpp:165]: This indirection is making the code more complex than simply copy-pasting the one or two lines to each location. Using the `run` variable to control the loop is fine but the `action` variable is pointless.

**Review by nextghost — CHANGES_REQUESTED** [review]

File: signus/src/mainmenu.cpp (hunk 1 above)
Comment (jefetienne) [reply signus/src/mainmenu.cpp]: Done!

File: signus/src/mainmenu.cpp (hunk 2 above)
Comment (jefetienne) [reply signus/src/mainmenu.cpp]: Done!

File: signus/src/mainmenu.cpp (hunk 3 above)
Comment (jefetienne) [reply signus/src/mainmenu.cpp]: Got it!

**Comment by nextghost** [comment]
Thanks, squashed and merged.

**Pull request merged**